pkg/apis/ceph.rook.io/v1/network.go

```diff
@@ -39,6 +39,9 @@ func (n *NetworkSpec) IsHost() bool {
 }
 
 func ValidateNetworkSpec(clusterNamespace string, spec NetworkSpec) error {
+	if spec.HostNetwork && (spec.Provider != NetworkProviderDefault) {
+		return errors.Errorf("the legacy hostNetwork setting is only valid with the default network provider'\"\"' and not with '%q'", spec.Provider)
```

Comment:
It reads more nicely to use Go's grave-accent string quotes when both double and single quotes are involved.

```suggestion
		return errors.Errorf(1the legacy hostNetwork setting is only valid with the default network provider ("") and not with '%q'`, spec.Provider)
```